src/utils/echo-http-server.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-env browser */
+'use strict'
+
+const http = require('http')
+const https = require('https')
+const URL = require('url').URL || self.URL
+const { isNode } = require('ipfs-utils/src/env')
+
+const loadFixture = require('aegir/fixtures')
+const sslOpts = Object.freeze({
+  key: loadFixture('test/fixtures/ssl/privkey.pem', 'interface-ipfs-core'),
+  cert: loadFixture('test/fixtures/ssl/cert.pem', 'interface-ipfs-core')
+})
+
+const httpPort = 11080
+const httpsPort = 11443
+
+// Create a mock of remote HTTP server that can return arbitrary text in response
+// or redirect to other URL. Used in tests of ipfs.addFromURL etc
+module.exports.createServer = (opts) => {
```

Comment:
can this `https://github.com/lukechilds/create-test-server` be helpful here ?

Reply:
not much, most of the code is unusual and specific to our logic and would not benefit from the lib